src/auth/ejabberd_auth.erl

```diff
@@ -463,6 +464,14 @@ do_get_passterm_with_authmodule(LUser, LServer) ->
 
 %% @doc Returns true if the user exists in the DB or if an anonymous user is
 %% logged under the given name
+-spec is_user_exists(JID :: jid:jid()) -> boolean().
+is_user_exists(#jid{luser = <<>>}) ->
```

Comment:
Since this is a new function can we call it `does_user_exist`? I think it is more correct.